tests/unit/test_create_json_schema.py

```diff
@@ -0,0 +1,1057 @@
+"""Tests for JSON Schema generation"""
+
+from typing import Generator, Any, Union, Optional
+import os
+import json
+from shutil import rmtree
+
+import pytest
+from jsonschema import Draft7Validator
+from jsonschema.exceptions import ValidationError
+
+from schematic.schemas.data_model_graph import DataModelGraphExplorer
+from schematic.schemas.create_json_schema import (
+    _get_ranges_from_range_rule,
+    _get_in_range_rule_from_rule_list,
+    _get_type_rule_from_rule_list,
+    JSONSchema,
+    Node,
+    GraphTraversalState,
+    create_json_schema,
+    _write_data_model,
+    _set_conditional_dependencies,
+    _set_property,
+    _create_enum_array_property,
+    _create_array_property,
+    _create_enum_property,
+    _create_simple_property,
+)
+from tests.utils import json_files_equal
+
+# pylint: disable=protected-access
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-positional-arguments
+
+
+@pytest.fixture(name="test_directory", scope="session")
+def fixture_test_directory(request) -> Generator[str, None, None]:
+    """Yields a directory for creating test jSON Schemas in"""
+    test_folder = "tests/data/json_schema_generator_output"
+
+    def delete_folder():
+        rmtree(test_folder)
```

Comment:
This is likely going to cause some potential race conditions. When using xdist to run tests even `session` scoped fixtures get copies made. In this case the first worker to finish is going to call `rmtree` to delete all data from the test folder. 

This should probably use a UUID4 string in the folder name to prevent collisions from concurrent tests.

Reply:
Oh that's interesting. I'll fix that.